File: date_today/perl_date_today/date_today.pl
#!/usr/bin/perl -w

use strict;
use warnings;

#sub date_today {
#  my ($sec,$min,$hour,$mday,$month,$year,$wday,$yday,$isdst) = localtime(time);
#  my @local = (localtime)[3..5];
#  my $date1 = sprintf("%04d-%02d-%02d", $local[2] + 1900, $local[1] + 1, $local[0]);
#  my @now = localtime(time());
#  ##my ($mday, $mon, $year) = (localtime(time))[3,4,5];
#  $month++;
#  $year = $year + 1900;
#  print "$year-$month-$mday $hour:$min:$sec\n";
#  #print $date1, "\n";
#  return $date1;
#}

sub date_today {
  my ($sec,$min,$hour,$mday,$month,$year,$wday,$yday,$isdst) = localtime(time);
  $month++;
  $year = $year + 1900;
  print "local=<$year-$month-$mday $hour:$min:$sec>\n";
  return "$year-$month-$mday $hour:$min:$sec";
}

sub date_today_utc {
  my ($sec,$min,$hour,$mday,$month,$year,$wday,$yday,$isdst) = gmtime(time);
  $month++;
  $year = $year + 1900;
  print "utc=<$year-$month-$mday $hour:$min:$sec>\n";
  return "$year-$month-$mday $hour:$min:$sec";
}

sub main {
  my $x;

  if( $#ARGV+1 != 0 ) {
    print STDERR "Usage: $0 <noargs>\n";
    exit(1);
  }

  print &date_today_utc(), "\n";
  print &date_today(), "\n";

  #print $ARGV[0], "\n";
  exit(0);
}

&main();

File: date_today/fsharp_date_today/date_today.fs
#light

open System

//comments

let date_utc =
  let utc = System.DateTime.UtcNow
  let year = utc.Year
  let month = utc.Month
  let day = utc.Day
  let hour = utc.Hour
  let min1 = utc.Minute
  let sec = utc.Second
  printfn "local=%A" (utc)
  sprintf "utc=<%A-%A-%A %A:%A:%A>" year month day hour min1 sec

let date_today =
  let local = System.DateTime.Now
  let year = local.Year
  let month = local.Month
  let day = local.Day
  let hour = local.Hour
  let min1 = local.Minute
  let sec = local.Second
  printfn "local=%A" (local)
  sprintf "local=<%A-%A-%A %A:%A:%A>" year month day hour min1 sec

[<EntryPoint>]
let main args =
  let argv = Environment.GetCommandLineArgs()

  if argv.Length <> 1 then
    eprintfn "Usage: %s <noargs>" argv.[0]
    -1
  else
    printfn "%s" date_utc
    printfn "%s" date_today
    0